intranet/apps/signage/views.py

```diff
@@ -26,3 +31,95 @@ def signage_display(request, display_id):
     context["sign"] = sign
     context["page_args"] = (sign, request)
     return render(request, "signage/base.html", context)
+
+
+def eighth(request):
+    block_id = request.GET.get("block_id")
+    block_increment = 0
+    internal_ip = check_internal_ip(request)
+    if internal_ip:
+        return internal_ip
+
+    if block_id is None:
+        block = EighthBlock.objects.get_first_upcoming_block()
+        if block is None:
+            last_block = EighthBlock.objects.order_by("date").last()
+            if last_block is not None:
+                block = last_block
+    else:
+        try:
+            block = (EighthBlock.objects.prefetch_related("eighthscheduledactivity_set").get(id=block_id))
```

Comment:
Why these parentheses?
Do we want to constrain getting a block from this school year only?

Reply:
Resolved in a force-push to fix the build.

Reply:
My second question still stands. Is it the intention of this page to only lookup blocks for this year or for any year?

Reply:
This page will only look up the next block within this year if not passed any special parameters, but it will allow looking up blocks for any year if requested.